lens-cube/src/main/java/org/apache/lens/cube/parse/DefaultQueryAST.java

```diff
@@ -68,4 +68,13 @@ public static DefaultQueryAST fromCandidateFact(CandidateFact fact, String stora
       ast.getFromString(),
       fact.getStorageWhereString(storageTable.substring(storageTable.indexOf(".") + 1)));
   }
+  public static DefaultQueryAST fromCandidateStorage(StorageCandidate sc, String storageTable, QueryAST ast) throws
+      LensException {
+    return new DefaultQueryAST(ast.getSelectAST(),
+        null,
+        ast.getGroupByAST(), ast.getHavingAST(), ast.getJoinAST(), ast.getOrderByAST(), ast.getLimitValue(),
+        ast.getFromString(),
+        //sc.getStorgeWhereStringMap().get(storageTable.substring(storageTable.indexOf(".") + 1)));
+        sc.getWhereString());
```

Comment:
Where string we are creating in CubeQueryContext and updated by other resolvers like StorageTableResolver etc. While creating default one we kept as null.